### PR TITLE
Inaugurate the Test Suite

### PR DESCRIPTION
**Pull request by CodaFi**
Boy, Swift fought tooth and nail for this not to happen.

### CODE REVIEW

File: AquiferTests/ProxyTests.swift [hunk 1]
```diff
@@ -6,6 +6,107 @@
...
+        
+        /// Need closures here.  Autoclosures crash Swiftc.
+        property("Request-Respond duality") <- forAll { (s : AServer, c : AClient) in
+            return
+                ({ $0.reflect() } • { respond($0) }, { request($0) }) ==== (s, c)
+                ^&&^
+                ({ $0.reflect() } • { request($0) }, { respond($0) }) ==== (s, c)
+        }
+
+        property("Request Zero Law") <- forAll { (p2 : AProxy, s : AServer, c : AClient) in
```
Comment (CodaFi) [inline AquiferTests/ProxyTests.swift:57]: We're failing this one going `>|>`, but passing going `<|<`.

File: AquiferTests/TestDefs.swift [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	typealias T = (Int -> Proxy<Int, Int, Int, Int, Int>)
+}
+
+struct Operation {
+	typealias T = (ProxyK.T, ProxyK.T) -> ProxyK.T
+}
+
+infix operator ==== {}
+
+func ==== (l : (ProxyK.T, ProxyK.T), r : (AServer, AClient)) -> Bool {
```
Comment (CodaFi) [inline AquiferTests/TestDefs.swift]: This needs to die.  Fancy Haskell tricks don't work very well here.

File: AquiferTests/ProxyTests.swift (hunk 1 above)
Comment (CodaFi) [inline AquiferTests/ProxyTests.swift:57]: Not anymore :tada:

**Comment by CodaFi** [comment]
I think this is a good start.

Ready for review.

**Comment by pthariensflame** [comment]
Looks good!

**Pull request merged**